app/models/manageiq/providers/openstack/network_manager/refresh_parser.rb

```diff
@@ -0,0 +1,375 @@
+# TODO: Separate collection from parsing (perhaps collecting in parallel a la RHEVM)
+
+module ManageIQ::Providers
+  class Openstack::NetworkManager::RefreshParser
+    include ManageIQ::Providers::Openstack::RefreshParserCommon::HelperMethods
+
+    def self.ems_inv_to_hashes(ems, options = nil)
+      new(ems, options).ems_inv_to_hashes
+    end
+
+    def initialize(ems, options = nil)
+      @ems               = ems
+      @connection        = ems.connect
+      @options           = options || {}
+      @data              = {}
+      @data_index        = {}
+      @known_flavors     = Set.new
+      @resource_to_stack = {}
+
+      @os_handle        = ems.openstack_handle
+      @network_service  = @os_handle.detect_network_service
+
+      validate_required_services
+    end
+
+    def validate_required_services
+      unless @network_service
+        raise MiqException::MiqOpenstackNetworkServiceMissing, "Required service Neutron is missing in the catalog."
+      end
+    end
+
+    def ems_inv_to_hashes
+      log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{@ems.name}] id: [#{@ems.id}]"
+
+      $fog_log.info("#{log_header}...")
+
+      get_security_groups
+      get_networks
+      get_network_routers
+      get_floating_ips
+      get_network_ports
+
+      $fog_log.info("#{log_header}...Complete")
+
+      link_network_ports_associations
+
+      @data
+    end
+
+    private
+
+    def parent_manager_fetch_path(collection, ems_ref)
+      @parent_manager_data ||= {}
+      return @parent_manager_data.fetch_path(collection, ems_ref) if @parent_manager_data.has_key_path?(collection, ems_ref)
+
+      @parent_manager_data.store_path(collection, ems_ref, @ems.public_send(collection).try(:where, :ems_ref => ems_ref).try(:first))
```

Comment:
If I'm reading this right, then I'm a little concerned with this.  Not for this PR specifically (I think), but for the larger implication of how we link network managers with other providers.

I can't put a finger on it yet, but I think we'll have to figure out how this linkage works longer term.

This might be fine for a first pass which is completely OSP specific.